src/audio/codec_adapter/codec_adapter.c

```diff
@@ -220,12 +220,21 @@ static int codec_adapter_prepare(struct comp_dev *dev)
 	return 0;
 }
 
+static int codec_adapter_params(struct comp_dev *dev,
+				    struct sof_ipc_stream_params *params)
+{
+	comp_dbg(dev, "codec_adapter_params(): codec_adapter doesn't support .params() method.");
+
+	return 0;
+}
```

Comment:
I don't like this commit that adds no functionality. The method is optional, so you might instead want to document that you want to support it in the future, rather than add a dummy implementation. params() will always be called on it if it is made available.